### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by amartya-rs**
### In this PR
- added filter by tags functionality
- some code refactoring
### Preview: https://deploy-preview-5--dopenotes.netlify.app/

### CODE REVIEW

**Review by archit7448 — APPROVED** [review]
amazing work ! 
code structure is proper 
UI looks good 
code looks good to merge

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Code is clean and easy to understand.

Good to merge 👍🏻

**Review by bhushannaikaroskar — APPROVED** [review]
Hey Amartya, awesome code structure and loved the way you have simplified code logic. Code is readable and simple and easily readable. Approving your merge.

File: src/components/filter-bar/filter-bar.css [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+   margin-left: auto;
+   margin-right: 2rem;
+}
+
+.filter-bar-wrapper .button {
+   margin-left: auto;
+}
+
+.filter-bar label {
+   font-weight: 500;
```
Comment (varun13169) [inline src/components/filter-bar/filter-bar.css]: To maintain consistency, preferably define it in typography and then use it.

File: src/pages/archive-page/ArchivePage.jsx [hunk 2]
```diff
@@ -34,21 +34,16 @@ const ArchivePage = () => {
...
-            },
-         });
-         dispatch({
-            type: "SAVE_ARCHIVED_NOTES_FROM_SERVER",
-            payload: response.data.archives,
-         });
-      } catch (error) {
-         console.log(error);
-      }
+   //moving note to trash
```
Comment (varun13169) [inline src/pages/archive-page/ArchivePage.jsx:37]: Nice comment explaining functionality

Comment (amartya-rs) [reply src/pages/archive-page/ArchivePage.jsx:37]: Thanks Varun!

File: src/components/filter-bar/filter-bar.css (hunk 1 above)
Comment (amartya-rs) [reply src/components/filter-bar/filter-bar.css]: Got your point.

**Pull request merged**